lib/tasks/test_security_helper.rb

```diff
@@ -24,6 +24,14 @@ def self.brakeman(format: "human")
     puts "**   engines:"
     puts "**   - #{engine_paths.join("\n**   - ")}"
 
+    # Brakeman's Gemfile detection does not work properly with engines
+    #   Brakeman detects the Gemfile.lock from the application root directory,
+    #   however when running from an engine the lockfile is in the engine
+    #   directory. So, we copy the Gemfile.lock into the application directory.
+    if defined?(ENGINE_ROOT)
+      FileUtils.cp(File.join(ENGINE_ROOT, "Gemfile.lock"), File.join(app_path, "Gemfile.lock"))
```

Comment:
Should we only copy it over if it doesn't exist previously?

```ruby
if defined?(ENGINE_ROOT) && !File.exist?(File.join(app_path, "Gemfile.lock"))
   FileUtils.cp(File.join(ENGINE_ROOT, "Gemfile.lock"), File.join(app_path, "Gemfile.lock"))
end
```

Not sure if we need to check `if File.exist?(File.join(ENGINE_ROOT, "Gemfile.lock"))`

Reply:
I'm concerned that an Engine run of the security suite could overwrite a symlinked manageiq in dev and possibly lead to weird issues if the engine has new dependencies.  If the core lockfile doesn't exist, that should less dev centered and hit the exact use case for CI.

Reply:
I debated that myself, and my conclusion was that there are 2 reasons you'd run locally

1. To actually test it (probably to duplicate what's in GHA), in which case the core Gemfile.lock is wrong unless it matches what's in the ui-classic directory, so we should overwrite it to be correct.
2. To add entries to the ignore list, in which case, again, the core Gemfile.lock is wrong and you must overwrite it anyway.

In both cases, not overwriting the core Gemfile.lock leads to incorrect results, which would lead to confusion.

Additionally, IMO, the core Gemfile.lock is ephemeral / rebuildable, so I'm more inclined to allow it to be overwritten, as opposed to, say, actual code changes or the Gemfile itself.

Reply:
I could print out a warning if it overwrites the file, if you think that's useful, but it will likely be WAY at the top of a very long list of output, and probably ignored. 🤷

Reply:
> Not sure if we need to check if File.exist?(File.join(ENGINE_ROOT, "Gemfile.lock"))

You can't even run the rake task without bundling first, so it's guaranteed to exist.

Reply:
Ok, in side conversation copying it off before overwriting would lead to a dirty git tree.  Let's see how this goes as is.